code/datums/martial/knifeboxing.dm

```diff
@@ -0,0 +1,56 @@
+/datum/martial_art/knifeboxing
+	name = "Knife-boxing"
+
+/datum/martial_art/knifeboxing/disarm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	to_chat(A, "<span class='warning'>Can't disarm while knife-boxing!</span>")
+	return TRUE
+
+/datum/martial_art/knifeboxing/grab_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	to_chat(A, "<span class='warning'>Can't grab while knife-boxing!</span>")
+	return TRUE
+
+/datum/martial_art/knifeboxing/harm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+
+	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+
+	var/atk_verb = pick("left hook","right hook","straight punch")
+
```

Comment:
Dont get why there are random spaces here

Reply:
i dunno i just ported code and it compiles